src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
 
+        if (key == null) {
+            putForNullKey(value);
+        } else if (table[hash(key)] == null) {
+            putOnEmptyBucket(key, value);
+        } else {
+            putWithCollision(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return table[0].value;
+        }
+
+        int index = hash(key);
+        Node<K, V> nodeInBucket = table[index];
+        while (nodeInBucket != null) {
+            if (key.equals(nodeInBucket.key)) {
+                return nodeInBucket.value;
+            }
+            nodeInBucket = nodeInBucket.nextNode;
+        }
```

Comment:
OPTIONAL: You can do this code without any extra checking
```suggestion
        int index = hash(key);
        Node<K, V> nodeInBucket = table[index];
        while (nodeInBucket != null) {
            if (Objects.equals(key, nodeInBucket.key)) {
                return nodeInBucket.value;
            }
            nodeInBucket = nodeInBucket.nextNode;
        }
```